Src/ResourceUploadBatch.cpp

```diff
@@ -394,11 +412,12 @@ class ResourceUploadBatch::Impl
         ThrowIfFailed(fence->SetEventOnCompletion(1ULL, gpuCompletedEvent));
 
         // Create a packet of data that'll be passed to our waiting upload thread
-        UploadBatch* uploadBatch = new UploadBatch();
+        auto uploadBatch = new UploadBatch();
         uploadBatch->CommandList = mList;
         uploadBatch->Fence = fence;
         uploadBatch->GpuCompleteEvent = gpuCompletedEvent;
-        uploadBatch->TrackedObjects.insert(std::end(uploadBatch->TrackedObjects), std::begin(mTrackedObjects), std::end(mTrackedObjects));
+        std::swap(mTrackedObjects, uploadBatch->TrackedObjects);
```

Comment:
Looking at the logic here, it makes more sense to ``swap`` than it does to copy then reset...